config/intellij-idea-inspections.xml

```diff
@@ -2594,6 +2594,7 @@
     <scope name="Tests" level="ERROR" enabled="false">
       <option name="ignoreInstanceofOnLibraryClasses" value="false"/>
     </scope>
+    <scope class="SiteUtil" level="ERROR" enabled="false"/>
```

Comment:
Explain this please

Reply:
Violation at

https://app.circleci.com/pipelines/github/checkstyle/checkstyle/19794/workflows/5625e8ee-9bc6-4358-8ecb-295367dc2619/jobs/343601/artifacts

```xml
<problems is_local_tool="true">
<problem>
<file>
file://$PROJECT_DIR$/src/main/java/com/puppycrawl/tools/checkstyle/site/ViolationMessagesMacro.java
</file>
<line>108</line>
<module>project</module>
<package>com.puppycrawl.tools.checkstyle.site</package>
<entry_point TYPE="method" FQNAME="com.puppycrawl.tools.checkstyle.site.ViolationMessagesMacro java.util.Set<java.lang.reflect.Field> getCheckMessageKeyFields(java.lang.Class<?> clss)"/>
<problem_class id="InstanceofChain" severity="ERROR" attribute_key="ERRORS_ATTRIBUTES">Chain of 'instanceof' checks</problem_class>
<description>
Chain of class equality checks indicates abstraction failure #loc
</description>
<highlighted_element>if</highlighted_element>
<language>JAVA</language>
<offset>12</offset>
<length>2</length>
</problem>
</problems>
```

It's talking about 
```java
            if (module == RegexpMultilineCheck.class) {
                checkstyleMessageKeys.addAll(getCheckMessageKeys(
                        Class.forName(
                                "com.puppycrawl.tools.checkstyle.checks.regexp.MultilineDetector"
                        ), checkInstance)
                );
            }
            else if (module == RegexpSinglelineCheck.class
                    || module == RegexpSinglelineJavaCheck.class) {
                checkstyleMessageKeys.addAll(getCheckMessageKeys(
                        Class.forName(
                                "com.puppycrawl.tools.checkstyle.checks.regexp.SinglelineDetector"
                        ), checkInstance)
                );
            }
```

These are edge cases in my opinion. Method copied from https://github.com/checkstyle/checkstyle/blob/34e7d5c714b2df62781790537c6127b5712ec097/src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/CheckUtil.java#L211

Reply:
We suppress inspections by javadoc tags, please suppress by it

Reply:
And this violation is right, it is hack.
We should have separate ticket to improve it by refactoring of modules to have  all MSG fields in module as field, all child, nested, composited classes should take by constructor content of message.
We already discussed it multiple times, and now we see real need for this

Reply:
Just to be clear: we suppress in javadoc `noinspection` tag by link to issue where we will do it in better way.

We will continue extend this if-else while converting more and more to use of this macros, and in separate PR will do optimization to avoid such hack.

Reply:
Opened at https://github.com/checkstyle/checkstyle/issues/13500

Reply:
Done. Suppressed.